src/ensembl/src/shared/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import defaultStyles from './Checkbox.scss';
+
+export enum CheckboxStatus {
+  CHECKED = 'checked',
+  DISABLED = 'disabled',
+  UNCHECKED = 'unchecked'
+}
+
+type Props = {
+  onClick?: () => void;
+  status: CheckboxStatus;
+  classNames?: { [key in CheckboxStatus]: string };
+  label?: string;
+  labelClassName?: string;
+};
+
+const Checkbox = (props: Props) => {
+  const [checkedStatus, toggleCheckedStatus] = useState(props.status);
+
+  const handleOnClick = () => {
+    // Do nothing if it is disabled
+    if (props.status === CheckboxStatus.DISABLED) {
+      return;
+    }
+
+    // Toggle the checked status
+    if (checkedStatus === CheckboxStatus.CHECKED) {
+      toggleCheckedStatus(CheckboxStatus.UNCHECKED);
+    } else {
+      toggleCheckedStatus(CheckboxStatus.CHECKED);
+    }
+
+    // Call the onClick function if we have one
+    if (props.onClick) {
+      props.onClick();
+    }
+  };
+
+  const styles = props.classNames
+    ? { ...defaultStyles, ...props.classNames }
+    : defaultStyles;
+
+  const className = classNames(styles.defaultCheckbox, styles[checkedStatus]);
+  const labelClassName = classNames(
+    defaultStyles.defaultLabel,
+    props.labelClassName
+  );
+
+  return (
+    <div onClick={handleOnClick} className={defaultStyles.checkboxHolder}>
+      <div className={className} />
+      {props.label && <div className={labelClassName}>{props.label}</div>}
+    </div>
```

Comment:
Perhaps better write this component using browser's native `label` and `input` tags?

(since `input` isn't properly styleable, it will have to be hidden via display: none, and styles will apply to a div/span element close to it, see e.g. discussion [here](https://stackoverflow.com/questions/4148499/how-to-style-a-checkbox-using-css))

Reply:
@azangru In the same thread, there is this answer: https://stackoverflow.com/a/32876109/8899157. It states that we should not use `display: none`. I think this is because of accessibility reasons. Maybe using negative `z-index` as suggested in that answer is a better approach?

Reply:
That's all done.